src/barril/units/_value_generator.py

```diff
@@ -56,12 +56,12 @@ def __iter__(self):
         elif self.iterate_1st and not self.iterate_2nd:
             static_value = self.p2
             for v in self.p1:
-                yield static_value, v
+                yield v, static_value
```

Comment:
Who else uses this file? I'm afraid they must also be missing tests.

Reply:
Hmm then we should also test the same operators for `Array`.

Please move your test to another module, say, `test_operators.py`, then we can parametrize the test also by array type:

```python
def MakeArray(values):
    """Creates a standard array used by testScalarOperations"""
    return units.Array("length", values, "m")    

def MakeFixedArray(values):
    """Creates a fixed array with 3 dimensions used by testScalarOperations"""
    return units.FixedArray(3, "length", values, "m")    

@pytest.mark.parametrize(
    "left, op, right, expected",
    [
        ([100, 80, 50], operator.truediv, 2, [50, 40, 25]),
        ([100, 80, 50], operator.sub, 2, [98, 78, 48]),
        (2, operator.sub, [100, 80, 50], [-98, -78, -48]),
        (2, operator.add, [100, 80, 50], [102, 82, 52]),
        ([100, 80, 50], operator.add, 2, [102, 82, 52]),
        (2, operator.mul, [100, 80, 50], [200, 160, 100]),
        ([100, 80, 50], operator.mul, 2, [200, 160, 100]),
    ],
)
@pytest.mark.parametrize('array_maker', [MakeArray, MakeFixedArray])
def testScalarOperations(left, op, right, expected, array_maker):
    if isinstance(left, list):
        left = array_maker(left)
    if isinstance(right, list):
        right = array_maker(right)

    result = op(left, right)
    assert result == array_maker(expected)
```

With just this we are now testing every implementation, and in case we add a new array-like implementation, it is just a matter of adding a new value to the `array_maker` parametrization. 😉